tests/test_controller.py

```diff
@@ -0,0 +1,304 @@
+
+	@pytest.mark.asyncio
+	async def test_error_handling(self, controller, browser_context):
+		"""Test error handling when an action fails."""
+		# Create an action with an invalid index
+		invalid_action = {'click_element_by_index': ClickElementAction(index=9999)}
+
+		class ClickActionModel(ActionModel):
+			click_element_by_index: Optional[ClickElementAction] = None
+
+		# This should fail since the element doesn't exist
+		with pytest.raises(Exception) as excinfo:
+			await controller.act(ClickActionModel(**invalid_action), browser_context)
+
+		# Verify that an appropriate error is raised
+		assert 'does not exist' in str(excinfo.value) or 'Element with index' in str(excinfo.value)
+
+	@pytest.mark.asyncio
+	async def test_wait_action(self, controller, browser_context):
+		"""Test that the wait action correctly waits for the specified duration."""
+		# Create wait action for 1 second - fix to use a dictionary
+		wait_action = {'wait': {'seconds': 1}}  # Corrected format
+
+		class WaitActionModel(ActionModel):
+			wait: Optional[dict] = None
+
+		# Record start time
+		start_time = time.time()
+
+		# Execute wait action
+		result = await controller.act(WaitActionModel(**wait_action), browser_context)
+
+		# Record end time
+		end_time = time.time()
+
+		# Verify the result
+		assert isinstance(result, ActionResult)
+		assert 'Waiting for' in result.extracted_content
+
+		# Verify that at least 1 second has passed
+		assert end_time - start_time >= 0.9  # Allow some timing margin
+
+	@pytest.mark.asyncio
+	async def test_go_back_action(self, controller, browser_context):
+		"""Test that go_back action navigates to the previous page."""
+		# Navigate to first page
+		goto_action1 = {'go_to_url': GoToUrlAction(url='https://google.com')}
+
+		class GoToUrlActionModel(ActionModel):
+			go_to_url: Optional[GoToUrlAction] = None
+
+		await controller.act(GoToUrlActionModel(**goto_action1), browser_context)
+
+		# Store the first page URL
+		page1 = await browser_context.get_current_page()
+		first_url = page1.url
+		print(f'First page URL: {first_url}')
+
+		# Navigate to second page
+		goto_action2 = {'go_to_url': GoToUrlAction(url='https://yahoo.com')}
+		await controller.act(GoToUrlActionModel(**goto_action2), browser_context)
+
+		# Verify we're on the second page
+		page2 = await browser_context.get_current_page()
+		second_url = page2.url
+		print(f'Second page URL: {second_url}')
+		assert 'yahoo.com' in second_url.lower()
+
+		# Execute go back action
+		go_back_action = {'go_back': NoParamsAction()}
+
+		class GoBackActionModel(ActionModel):
+			go_back: Optional[NoParamsAction] = None
+
+		result = await controller.act(GoBackActionModel(**go_back_action), browser_context)
+
+		# Verify the result
+		assert isinstance(result, ActionResult)
+		assert 'Navigated back' in result.extracted_content
+
+		# Add another delay to allow the navigation to complete
+		await asyncio.sleep(1)
```

Comment:
During testing, I discovered the `go_back` function in the browser_use\browser\context.py file has a very short timeout (10ms) when navigating browser history. This caused test failures as the navigation wasn't complete before assertions ran.

for temporary, the test now includes a 1-second sleep after back navigation to ensure reliable page loading before making assertions. A better long-term solution would be to improve the core implementation with proper wait handling. but if just increase the timeout, then this sleep is not required.